File: backend/src/main/java/com/ehr/staffservice/service/ServiceService.java
package com.ehr.staffservice.service;

import com.ehr.staffservice.dto.ServiceDto;
import java.util.List;

public interface ServiceService {
    ServiceDto create(ServiceDto dto);
    ServiceDto update(Long id, ServiceDto dto);
    ServiceDto get(Long id);
    List<ServiceDto> getAll();
    void delete(Long id);
}

File: frontend/src/app/core/services/service.service.ts
import { Injectable } from '@angular/core';
import { ApiService } from './api.service';
import { Observable } from 'rxjs';
import { Service, CreateServiceDto } from '../models/service.model';

@Injectable({
  providedIn: 'root'
})
export class ServiceService {
  constructor(private apiService: ApiService) {}

  getAll(): Observable<Service[]> {
    return this.apiService.get<Service[]>('/api/services');
  }

  getById(id: number): Observable<Service> {
    return this.apiService.get<Service>(`/api/services/${id}`);
  }

  create(service: CreateServiceDto): Observable<Service> {
    return this.apiService.post<Service>('/api/services', service);
  }

  update(id: number, service: Partial<CreateServiceDto>): Observable<Service> {
    return this.apiService.put<Service>(`/api/services/${id}`, service);
  }

  delete(id: number): Observable<void> {
    return this.apiService.delete<void>(`/api/services/${id}`);
  }
}

